src/options.ts

```diff
@@ -5,7 +5,22 @@ import type { WatchOptions } from 'chokidar'
  * @param filename absolute path to the file
  * @returns the transformed content. when `null` is returned, the file won't be created.
  */
-export type TransformFunc = (content: string, filename: string) => string | null
+type TransformFunc<T extends string | Buffer> = (
+  content: T,
+  filename: string
+) => T | null
+
+export type TransformOptionObject =
+  | {
+      encoding: Exclude<BufferEncoding, 'binary'>
+      handler: TransformFunc<string>
+    }
+  | {
+      encoding: 'binary'
+      handler: TransformFunc<Buffer>
+    }
```

Comment:
```suggestion
  | {
      encoding: Exclude<BufferEncoding, 'binary'>
      handler: TransformFunc<string>
    }
  | {
      encoding: 'buffer'
      handler: TransformFunc<Buffer>
    }
```
Now I think this is better because `'binary'` means binary string.

Reply:
done